pykechain/models/property_reference.py

```diff
@@ -127,8 +127,9 @@ def _validate_values(self) -> List[Text]:
 
         object_ids = []
         for value in self._value:
-            if isinstance(value, dict) and "pk" in value:
-                object_ids.append(str(value.get("pk")))
+            if isinstance(value, dict) and "pk" in value or "id" in value:
+                pk = str(value.get("pk", value.get("id")))
+                object_ids.append(pk)
             elif isinstance(value, (int, str)):
                 object_ids.append(str(value))
```

Comment:
What if this is not an UUID? I know it's unlikely, but maybe we should see if the string is an UUID first?

Reply:
That is the problem with user references, they are not defined by UUIDs but integers, hence this separate function. You will find the same method in the `BaseReference` class, where the key is `id` instead of `pk`.

In addition, this `_validate_values()` is an internal method that is already type-checked if set via `.value` or `part.update()`, or is retrieved from KE-chain directly and requires no additional validation.

Reply:
To elaborate, the reason I added the `id` here as well is to:

1. support the creation of a dict with key `id` in the `value` method of the BaseReference class
2. Forwards compatibility in case KE-chain makes the user data homogenous with the other reference properties

Reply:
Yeah, you're right, didn't think about the `UserReference` properties